### PR TITLE
Add genhackage

### PR DESCRIPTION
**Pull request by felixonmars**
Tool to submit Arch package info to hackage and convert Cabal library
name to Arch package name.

### CODE REVIEW

File: package/genhackage [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+    result_fd.seek(0)
+    result_body = result_fd.read().rstrip("\r\n")
+    return result_body
+
+
+def submit_hackage(repo):
+    csv_body = gen_hackage_csv(repo)
+    headers={"Content-Type": "text/csv"}
+    if "HACKAGE_USERNAME" in os.environ and "HACKAGE_PASSWORD" in os.environ:
```
Comment (Foxboron) [inline package/genhackage]: should the HACKAGE env variables be documented somewhere in the header of the file?

File: package/genhackage [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+
+# SPDX-License-Identifier: GPL-2.0
+
+# Please set HACKAGE_API_TOKEN or both HACKAGE_USERNAME and HACKAGE_PASSWORD to submit package list
+# to hackage!
+
+import os
+import csv
+from io import StringIO
+from sh import pacman, expac
```
Comment (SantiagoTorres) [inline package/genhackage]: I wonder if we'd like to document the non stdlib dependencies somehow , in this case I think `python-requests` and `python-sh`

Comment (Foxboron) [reply package/genhackage]: optdepends in the package :)?

Comment (felixonmars) [reply package/genhackage]: And also `expac`,

Comment (SantiagoTorres) [reply package/genhackage]: Yeah that's definitely one way, but it'd be nice to have it in the PR :F

File: package/genhackage [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        if package:
+            if not quiet:
+                print("Found", package, version)
+
+            version_mapping[package] = version
+            arch_package_mapping[package] = "ghc"
+
+            counter += 1
+
+    for line in pacman("-Sl", repo, "--color=never"):
```
Comment (jelly) [inline package/genhackage]: Hmm could use pyalpm maybe?

**Comment by felixonmars** [comment]
Abandoned as the functionality has been merged into arch-hs in 0.2.0.0.

**Pull request closed without merging**